scottypy/app.py

```diff
@@ -114,18 +115,22 @@ def _list(beam):
         _list(scotty.get_beam(beam_id_or_tag))
 
 
-def _download_beam(beam, dest, overwrite, filter):
+
+async def _download_beam(beam, dest, overwrite, filter):
     if not os.path.isdir(dest):
         os.makedirs(dest)
 
     click.echo("Downloading beam {} to directory {}".format(beam.id, dest))
 
+    tasks = []
     for file_ in beam.get_files(filter_=filter):
         click.echo("Downloading {}".format(file_.file_name))
         try:
-            file_.download(dest, overwrite=overwrite)
+            task = asyncio.ensure_future(file_.download(dest, overwrite=overwrite))
+            tasks.append(task)
         except NotOverwriting as e:
             click.echo("{} already exists. Use --overwrite to overwrite".format(e.file))
+    await asyncio.gather(*tasks, return_exceptions=True)
```

Comment:
Exceptions aren't handled.

Reply:
How should we handle them?

Reply:
Either raise them or go over the returned exceptions and do something with them.